ddtrace/propagation/http.py

```diff
@@ -150,6 +150,55 @@ class _DatadogMultiHeader:
     def _is_valid_datadog_trace_tag_key(key):
         return key.startswith("_dd.p.")
 
+    @staticmethod
+    def _get_tags_value(headers):
+        # type: (Dict[str, str]) -> Optional[str]
+        return _extract_header_value(
+            _POSSIBLE_HTTP_HEADER_TAGS,
+            headers,
+            default="",
+        )
+
+    @staticmethod
+    def _extract_meta(tags_value):
+        # Do not fail if the tags are malformed
+        try:
+            meta = {
+                k: v
+                for (k, v) in decode_tagset_string(tags_value).items()
+                if (
+                    k not in _DatadogMultiHeader._X_DATADOG_TAGS_EXTRACT_REJECT
+                    and _DatadogMultiHeader._is_valid_datadog_trace_tag_key(k)
+                )
+            }
+        except TagsetMaxSizeDecodeError:
+            meta = {
+                "_dd.propagation_error": "extract_max_size",
+            }
+            log.warning("failed to decode x-datadog-tags", exc_info=True)
+        except TagsetDecodeError:
+            meta = {
+                "_dd.propagation_error": "decoding_error",
+            }
+            log.debug("failed to decode x-datadog-tags: %r", tags_value, exc_info=True)
+        return meta
+
+    @staticmethod
+    def _put_together_trace_id(trace_id_hob_hex, low_64_bits):
+        # combine highest and lowest order hex values to create a 128 bit trace_id
+        return int(trace_id_hob_hex + "{:016x}".format(low_64_bits), 16)
+
+    @staticmethod
+    def _higher_order_is_valid(upper_64_bits):
+        # type: (str) -> bool
```

Comment:
Using Python 3 type annotations makes the code more future-proof and compatible with typechecking tools.

Reply:
➕